internal/servers/controller/handlers/targets/tcp/target_service_test.go

```diff
@@ -2842,6 +2842,237 @@ func TestAuthorizeSession(t *testing.T) {
+	at := authtoken.TestAuthToken(t, conn, kms, org.GetPublicId())
+	ctx = auth.NewVerifierContext(requests.NewRequestContext(ctx),
+		iamRepoFn,
+		atRepoFn,
+		serversRepoFn,
+		kms,
+		&authpb.RequestInfo{
+			Token:       at.GetToken(),
+			TokenFormat: uint32(auth.AuthTokenTypeBearer),
+			PublicId:    at.GetPublicId(),
+		})
+
+	r := iam.TestRole(t, conn, proj.GetPublicId())
+	_ = iam.TestUserRole(t, conn, r.GetPublicId(), at.GetIamUserId())
+	_ = iam.TestRoleGrant(t, conn, r.GetPublicId(), "id=*;type=*;actions=*")
+
+	s, err := targets.NewService(ctx, kms, repoFn, iamRepoFn, serversRepoFn, sessionRepoFn, pluginHostRepoFn, staticHostRepoFn, credentialRepoFn)
+	require.NoError(t, err)
+
+	hc := static.TestCatalogs(t, conn, proj.GetPublicId(), 1)[0]
+	h := static.TestHosts(t, conn, hc.GetPublicId(), 1)[0]
+	shs := static.TestSets(t, conn, hc.GetPublicId(), 1)[0]
+	_ = static.TestSetMembers(t, conn, shs.GetPublicId(), []*static.Host{h})
+
+	v := vault.NewTestVaultServer(t)
+	v.AddKVPolicy(t)
+	sec, tok := v.CreateToken(t, vault.WithPolicies([]string{"default", "boundary-controller", "secret"}))
+
+	store := vault.TestCredentialStore(t, conn, wrapper, proj.GetPublicId(), v.Addr, tok, sec.Auth.Accessor)
+	credService, err := credentiallibraries.NewService(credentialRepoFn, iamRepoFn)
+	require.NoError(t, err)
+	mySecret := v.CreateKVSecret(t, "my-secret", []byte(`{"data": {"username": "my-user", "password": "my-pass"}}`))
+	require.NotNil(t, mySecret)
+	clsRespUserPassword, err := credService.CreateCredentialLibrary(ctx, &pbs.CreateCredentialLibraryRequest{Item: &credpb.CredentialLibrary{
+		CredentialStoreId: store.GetPublicId(),
+		Name:              wrapperspb.String("Userpassword Library"),
+		Description:       wrapperspb.String("Userpassword Library Description"),
+		Attributes: &structpb.Struct{Fields: map[string]*structpb.Value{
+			"path":        structpb.NewStringValue(path.Join("secret", "data", "my-secret")),
+			"http_method": structpb.NewStringValue("GET"),
+		}},
+		CredentialType: "user_password",
+	}})
+	require.NoError(t, err)
+	validateUserpassFn := func(t *testing.T, m map[string]interface{}) {
+		require.Len(t, m, 2)
+		user, ok := m["username"]
+		require.True(t, ok)
+		assert.Equal(t, "my-user", user)
+		pass, ok := m["password"]
+		require.True(t, ok)
+		assert.Equal(t, "my-pass", pass)
+	}
+
+	clsRespUnspecified, err := credService.CreateCredentialLibrary(ctx, &pbs.CreateCredentialLibraryRequest{Item: &credpb.CredentialLibrary{
+		CredentialStoreId: store.GetPublicId(),
+		Name:              wrapperspb.String("Unspecified Library"),
+		Description:       wrapperspb.String("Unspecified Library Description"),
+		Attributes: &structpb.Struct{Fields: map[string]*structpb.Value{
+			"path":        structpb.NewStringValue(path.Join("secret", "data", "my-secret")),
+			"http_method": structpb.NewStringValue("GET"),
+		}},
+	}})
+	require.NoError(t, err)
+	validateUnspecifiedFn := func(t *testing.T, m map[string]interface{}) {
```

Comment:
Since this and the `validateUserpassFn` are only used once consider inlining it so I don't have to jump to so many different parts of the file to see what input should result in what output.

Reply:
Inlined the unspecified but reusing the `validateUserpassFn` now